pkg/sypgp/sypgp.go

```diff
@@ -612,6 +619,140 @@ func SearchPubkey(search, keyserverURI, authToken string) error {
 	return nil
 }
 
+//function to obtain the algorithm name for key encryption
+func getEncryptionAlgorithmName(n string) (string, error) {
+	algorithmName := ""
+
+	code, err := strconv.ParseInt(n, 10, 64)
+	if err != nil {
+		return "", err
+	}
+	switch code {
+
+	case 1, 2, 3:
+		algorithmName = "RSA"
+	case 16:
+		algorithmName = "Elgamal"
+	case 17:
+		algorithmName = "DSA"
+	case 18:
+		algorithmName = "Elliptic Curve"
+	case 19:
+		algorithmName = "ECDSA"
+	case 20:
+		algorithmName = "Reserved"
+	case 21:
+		algorithmName = "Diffie-Hellman"
+	}
+	return algorithmName, nil
+}
+
+//function to obtain a date format from linux epoch time
+func date(s string) string {
+	ret := "NULL"
+	if s == "" {
+		return ret
+	}
+	if s == "none" {
+		return s + "		  	       "
+	}
+
+	c, _ := strconv.ParseInt(s, 10, 64)
```

Comment:
I don't think it's a good idea to ignore `err` here. When `ParseInt` fails:

> If s is empty or contains invalid digits, err.Err = ErrSyntax and the returned value is 0; if the value corresponding to s cannot be represented by a signed integer of the given size, err.Err = ErrRange and the returned value is the maximum magnitude integer of the appropriate bitSize and sign.

This will make for some interesting behaviour when these `c` values are interpreted by `time.Unix`. I'd strongly prefer to raise the error to the caller when invalid input is detected.